File: ProjectPos.FrontOffice.FrontEnd/src/proxy/interfaces/contact-person-dto.ts
import { FullAuditedEntityDto } from "../entity-dtos/full-audited-entity-dto";
import { JobTitle } from "../enums/job-title";
import { Title } from "../enums/title";
import { AddressDto } from "./address-dto";
import { CompanyDto } from "./company-dto";

export interface ContactPersonDto extends FullAuditedEntityDto<number> {
    companyId?: number;
    addressId: number;
    firstName?: string;
    lastName?: string;
    title?: Title;
    jobPosition?: JobTitle;
    phone1?: string;
    phone2?: string;
    email?: string;
    company?: CompanyDto;
    address?: AddressDto;
}

File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/ContactPersonDto.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using ProjectPos.Services.EntityDtos;
using ProjectPos.Data.Shared.Enums;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ProjectPos.Services.DTOs
{
    public class ContactPersonDto : AuditedEntityDto<int>
    {
        public int? CompanyId { get; set; }
        public int AddressId { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public Title? Title { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public JobTitle? JobPosition { get; set; }
        public string? Phone1 { get; set; }
        public string? Phone2 { get; set; }
        public string? Email { get; set; }

        [ForeignKey("CompanyId")]
        public CompanyDto? Company { get; set; }

        [ForeignKey("AddressId")]
        public AddressDto? Address { get; set; }
    }
}